main.go

```diff
@@ -426,14 +426,16 @@ func Flash(pkgName, port string, options *compileopts.Options) error {
 	// do we need port reset to put MCU into bootloader mode?
 	if config.Target.PortReset == "true" && flashMethod != "openocd" {
 		port, err := getDefaultPort(port, config.Target.SerialPort)
-		if err == nil {
-			err = touchSerialPortAt1200bps(port)
-			if err != nil {
-				return &commandError{"failed to reset port", port, err}
-			}
-			// give the target MCU a chance to restart into bootloader
-			time.Sleep(3 * time.Second)
+		if err != nil {
```

Comment:
This particular change may be questionable... before, if the device was already in a UF2 bootloader, flashing would proceed silently - with this change it would fail.

Perhaps should do the volume check first, and skip the reset if the volume already exists?

Reply:
> perhaps should do the volume check first, and skip the reset if the volume already exists to separate these cases

That sounds like a very good idea.

Reply:
I'm thinking for now, I'll just back out this part which causes a regression - i think the rest of the change is good.  We can revisit this bit at a later date.